arm-network/2017-03-01/swagger/networkWatcher.json

```diff
@@ -1714,6 +1769,176 @@
           "$ref": "#/definitions/FlowLogProperties"
         }
       }
+    },
+    "ConnectivityParameters": {
+      "description": "Parameters that determine how the connectivity check will be performed.",
+      "required": [
+        "source",
+        "destination"
+      ],
+      "properties": {
+        "source": {
+          "$ref": "#/definitions/ConnectivitySource"
+        },
+        "destination": {
+          "$ref": "#/definitions/ConnectivityDestination"
+        }
+      }
+    },
+    "ConnectivitySource": {
+      "description": "Parameters that define the source of the connection.",
+      "required": [
+        "resourceId"
+      ],
+      "properties": {
+        "resourceId": {
+          "description": "The ID of the resource from which a connectivity check will be initiated.",
+          "type": "string"
+        },
+        "port": {
+          "description": "The source port from which a connectivity check will be performed.",
+          "type": "integer"
+        }
+      }
+    },
+    "ConnectivityDestination": {
+      "description": "Parameters that define destination of connection.",
+      "properties": {
```

Comment:
no properties are required? So `destination` in `ConnectivityParameters` is required, but then it can be an empty object? What does your service do in that case?